spec/support/reporters/test-coverage.mjs

```diff
@@ -0,0 +1,23 @@
+import fsPromises from 'node:fs/promises';
+import {URL} from 'url';
+import path from 'node:path';
+
+export default class TestCoverageReporter {
+  suiteDone({properties}) {
+    this.coverage = properties?.__coverage__ || this.coverage;
+  }
+
+  jasmineDone() {
+    const thisFileDirname = path.dirname(new URL(import.meta.url).pathname);
+    const rootDirname = path.join(thisFileDirname, '..', '..', '..');
+    const reportPath = path.join(rootDirname, 'coverage', 'javascript');
+    const nycOutputPath = path.join(rootDirname, '.nyc_output');
+    const assetsPath = path.join(nycOutputPath, 'assets');
+    const testCoveragePath = path.join(nycOutputPath, 'out.json');
+    return Promise.all([
```

Comment:
Remove previous report and nyc-instrumented assets as they're not useful anymore and by removing them we make it more error-prone to run it multiple times, as they files related to the previous run are removed.

Reply:
This seems like really useful information, should it be a code comment so that it gets retained for the next developer?

Reply:
Hmm, I wonder, if a code comment is the best idea. Could it be documented somewhere else? As this code will be shared between repos in the future, what would be the best place? Where would you search for this kind of info?

Reply:
*sorry, I meant **less** error-prone of course

Reply:
As we discussed, having this as a code comment probably makes the most sense right now. Although it's possible that in the future we would have some designated place for docs once we share the functionality between repos.

Maybe we could shorten the code comment to say something like: 

```
// Remove the previous report and the nyc-instrumented assets as they are not needed after the run has completed.
```

Reply:
Done